### ISSUE TITLE
WAF Harvester

### ISSUE THREAD

**Issue by kimwdavidson** [GSA/datagov-ckan-multi#307]
## User Story
As a federal agency's data publisher, I want to describe my data in one place using federal standards and see it published on data.gov, so that I can comply with federal mandates in the fewest steps.

## Acceptance Criteria
- [x] GIVEN I add a GSA WAF harvest source
  WHEN I trigger a harvest run of that source
  THEN the harvest is successful and the metadata is ingested and visible on the [GSA organization page](https://catalog-next.sandbox.datagov.us/organization/gsa-gov)
  ~AND generates an email to publisher~ (covered in #315)

## Details / tasks
 - [x] Use the import tool to create a list of WAF sources and share with the team. Done [here](https://docs.google.com/spreadsheets/d/1LgVvgiUC5Vc-OQaSmtvQNlTFLhs3Mho65ekggYEiYh4/edit?ts=5e4177f8#gid=1535359331)
 - [x] Run manual tests and create new issues for the unexpected errors.
   + Created [#39](https://github.com/GSA/catalog.data.gov/issues/39)
   + Created [#43](https://github.com/GSA/catalog.data.gov/issues/43)
   + Created [#46](https://github.com/GSA/catalog.data.gov/issues/46)
 - [ ] Deploy to sandbox/staging the agreed WAF harvester source scope and QA the work.
 - [ ] The WAF harvester is part of the _spatial_ extension. We are moving to upstream this extension. We need to define is a code diff analysis is required - _On reflection, we suggest this should be in another issue_
 - [x] Check if WAF sources require further testing and add new issue to cover it
   + We have test for `GeminiWafHarvester` which is a type of `WafHarvester`. We can consider tests covered for these sources.

**Comment by thejuliekramer** [comment]
WAF sources to test https://docs.google.com/spreadsheets/d/1LgVvgiUC5Vc-OQaSmtvQNlTFLhs3Mho65ekggYEiYh4/edit?ts=5e4177f8#gid=1535359331

**Comment by thejuliekramer** [comment]
waf sources are importing in staging https://catalog-next.sandbox.datagov.us/harvest

**Comment by kimwdavidson** [comment]
@avdata99 are the tickets you mentioned above the blockers to this ticket?

**Comment by avdata99** [comment]
> @avdata99 are the tickets you mentioned above the blockers to this ticket? 

Yes @kimwdavidson. I didn't find a way to set this issues as blockers.

**Comment by kimwdavidson** [comment]
Blocked. Ready for QA when sandbox is ready.

**Comment by avdata99** [comment]
**QA**

**We can harvest WAF**

![image.png](https://images.zenhubusercontent.com/5d0fa29f8356d802835f2561/d58d1ed6-b58e-41af-952d-1f7a41d9b4d3)

**We can see datasets**

![image.png](https://images.zenhubusercontent.com/5d0fa29f8356d802835f2561/f88dfcd8-cb8c-486c-b1a4-15df83007768)

**We can see resources**

![image.png](https://images.zenhubusercontent.com/5d0fa29f8356d802835f2561/0f0113b2-a139-4481-83da-e9118b005876)

I think this is done

**Comment by adborden** [comment]
QA is possibly blocked by https://github.com/GSA/datagov-ckan-multi/issues/321

**Comment by adborden** [comment]
We discussed the upstream behavior of ckanext-harvester vs our fork. The Harvest Sources tab and the controller serving that is a data.gov customization. For now, this behavior isn't essential to QA, so we'll go ahead.

I created a new harvest source from https://catalog-next.sandbox.datagov.us/harvest/new: https://catalog-next.sandbox.datagov.us/harvest/wv-gis-technical-center/job/f56ccbe7-fe15-4b8e-994a-bfb70df068b4

Unfortunately, it ran into this error https://github.com/GSA/datagov-ckan-multi/issues/323 and could not complete successfully.

I'll look for another WAF source. It would be worth identifying these sources ahead of time and including them in the AC.
